src/autowiring/test/PostConstructTest.cpp

```diff
@@ -156,6 +198,15 @@ TEST_F(PostConstructTest, MultiNotifyWhenAutowired) {
   ASSERT_EQ(10, field) << "Autowiring lambdas did not run the expected number of times";
 }
 
+/*
+ Compile time failure case:
+class ForwardDeclared;
+TEST_F(PostConstructTest, TestForwardDeclare) {
+  Autowired<ForwardDeclared> decl;
+  decl.NotifyWhenAutowired([]() {std::cout << "thing"; });
+}
+*/
```

Comment:
Oh, I missed the part where you commented about what the failure should be.

I've thought about this, I experimented with CMake's `try_compile` feature, but decided against using that for things that aren't build environment provisioning.